Fix compile errors

seunghyeoks opened the pull request:
Hello!! I tried SGFuzz today, and I encountered some problems when i ran `build.sh`

What I changed is..

1. Update build.sh to use LF line break, because CRLF is not working on Linux & macOS.
2. Update FuzzerStateMachine.cpp to match the declaration in the header file.
3. Add .gitignore to avoid the built binary libsfuzzer.a from being included in git.

Looking forward to your review, thank you!

seunghyeoks committed:
chore: Changed to LF so that it can be run on Windows, Mac, and Linux

seunghyeoks committed:
fix: Modify the return type of the sum_hits() function to size_t to ensure consistency with the header declaration.

seunghyeoks committed:
chore: Add .gitignore to ignore libsfuzzer.a from git

bajinsheng commented:
Thanks for your contribution!
However, not sure about the first point, as the project was developed and tested on Linux Ubuntu.

seunghyeoks committed:
Revert "chore: Changed to LF so that it can be run on Windows, Mac, and Linux"

This reverts commit 123ad204c7e4272b6a7396b0dd3c85ac29772d42.

seunghyeoks commented:
Well, when I ran build.sh after cloning the code, the following error occurred.

<img width="932" height="635" alt="스크린샷 2025-09-26 오후 6 16 35" src="https://github.com/user-attachments/assets/c0ac2fca-a98f-4bdd-ab34-11ab68c940de" />


I noticed that build.sh was using ^M as the line break.

<img width="932" height="635" alt="스크린샷 2025-09-26 오후 6 16 17" src="https://github.com/user-attachments/assets/04c70563-5a88-4093-9e95-c111b83d37d8" />


However, what I discovered today is that even after I changed the line break to LF, pushed the commit, and cloned the repository again, it was still using ^M. So I assume there must be another reason for this issue.

I have reverted the commit that edited build.sh.
